RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -133,1112 +137,446 @@ void AntiElectronIDMVA6::beginEvent(const edm::Event& evt, const edm::EventSetup
-                                    Float_t ElecEtotOverPin,
-                                    Float_t ElecChi2NormGSF,
-                                    Float_t ElecChi2NormKF,
-                                    Float_t ElecGSFNumHits,
-                                    Float_t ElecKFNumHits,
-                                    Float_t ElecGSFTrackResol,
-                                    Float_t ElecGSFTracklnPt,
-                                    Float_t ElecPin,
-                                    Float_t ElecPout,
-                                    Float_t ElecEecal,
-                                    Float_t ElecDeltaEta,
-                                    Float_t ElecDeltaPhi,
-                                    Float_t ElecMvaInSigmaEtaEta,
-                                    Float_t ElecMvaInHadEnergy,
-                                    Float_t ElecMvaInDeltaEta) {
+template <class TauType, class ElectronType>
+double AntiElectronIDMVA6<TauType, ElectronType>::MVAValue(const TauVars& tauVars,
+                                                           const TauGammaVecs& tauGammaVecs,
+                                                           const ElecVars& elecVars) {
+  TauGammaMoms tauGammaMoms;
   double sumPt = 0.;
   double dEta2 = 0.;
   double dPhi2 = 0.;
-  double sumPt2 = 0.;
-  for (unsigned int i = 0; i < GammasPtInSigCone.size(); ++i) {
-    double pt_i = GammasPtInSigCone[i];
-    double phi_i = GammasdPhiInSigCone[i];
-    if (GammasdPhiInSigCone[i] > M_PI)
-      phi_i = GammasdPhiInSigCone[i] - 2 * M_PI;
-    else if (GammasdPhiInSigCone[i] < -M_PI)
-      phi_i = GammasdPhiInSigCone[i] + 2 * M_PI;
-    double eta_i = GammasdEtaInSigCone[i];
+  tauGammaMoms.signalPFGammaCandsIn = tauGammaVecs.gammasPtInSigCone.size();
+  for (size_t i = 0; i < tauGammaVecs.gammasPtInSigCone.size(); ++i) {
+    double pt_i = tauGammaVecs.gammasPtInSigCone[i];
+    double phi_i = tauGammaVecs.gammasdPhiInSigCone[i];
+    if (tauGammaVecs.gammasdPhiInSigCone[i] > M_PI)
+      phi_i = tauGammaVecs.gammasdPhiInSigCone[i] - 2 * M_PI;
+    else if (tauGammaVecs.gammasdPhiInSigCone[i] < -M_PI)
+      phi_i = tauGammaVecs.gammasdPhiInSigCone[i] + 2 * M_PI;
+    double eta_i = tauGammaVecs.gammasdEtaInSigCone[i];
     sumPt += pt_i;
-    sumPt2 += (pt_i * pt_i);
     dEta2 += (pt_i * eta_i * eta_i);
     dPhi2 += (pt_i * phi_i * phi_i);
   }
-  Float_t TauGammaEnFracIn = -99.;
-  if (TauPt > 0.) {
-    TauGammaEnFracIn = sumPt / TauPt;
+
+  tauGammaMoms.gammaEnFracIn = -99.;
+  if (tauVars.pt > 0.) {
+    tauGammaMoms.gammaEnFracIn = sumPt / tauVars.pt;
   }
   if (sumPt > 0.) {
     dEta2 /= sumPt;
     dPhi2 /= sumPt;
   }
-  Float_t TauGammaEtaMomIn = std::sqrt(dEta2) * std::sqrt(TauGammaEnFracIn) * TauPt;
-  Float_t TauGammaPhiMomIn = std::sqrt(dPhi2) * std::sqrt(TauGammaEnFracIn) * TauPt;
+  tauGammaMoms.gammaEtaMomIn = std::sqrt(dEta2) * std::sqrt(tauGammaMoms.gammaEnFracIn) * tauVars.pt;
```

Comment:
In case of
```
   tauGammaMoms.gammaEtaMomIn = std::sqrt(dEta2*tauGammaMoms.gammaEnFracIn) * tauVars.pt;
```
you can save one expensive ```sqrt``` operation (same below)